### PR TITLE
Rust: Migrate to Prost

### PR DESCRIPTION
**Pull request by nrc**
This PR makes tipb support both Prost and rust-protobuf as codecs.

PTAL @overvenus

### CODE REVIEW

File: include/gogoproto/gogo.proto [hunk 1]
```diff
@@ -0,0 +1,136 @@
+// Protocol Buffers for Go with Gadgets
```
Comment (Hoverbear) [inline include/gogoproto/gogo.proto]: Was this included on purpose for prost?

Comment (nrc) [reply include/gogoproto/gogo.proto]: Yeah, iirc Prost had errors without having this proto here. I'll update the README

Comment (nrc) [reply include/gogoproto/gogo.proto]: Hmm, I'm not clear on what the README is saying. It looks to me like it is just installing tools, not the protobuf specs, but I don't know the Go/protobuf situation.

Comment (nrc) [reply include/gogoproto/gogo.proto]: ping @Hoverbear do you think the README is saying something else?

Comment (nrc) [reply include/gogoproto/gogo.proto]: ping @Hoverbear PTAL at the above questions

**Comment by hicqu** [comment]
I'm not permitted to merge it. :(

**Comment by nrc** [comment]
ping @AndreMouche could you merge this please? Nobody seems to have the permission :-p

File: include/gogoproto/gogo.proto (hunk 1 above)
Comment (Hoverbear) [reply include/gogoproto/gogo.proto]: I'm so sorry I didn't see this nick. This email thread was buried behind a pagination. If I'm being so delinquent again please ping me is a more direct way so I can catch it. 🙇‍♀ 

Let's leave it as is.

**Pull request merged**